### PR TITLE
[Fix] Show blank cell for undefined cell

### PR DESCRIPTION
**Pull request by kubakrupski**
### Description
Show blank cell for undefined cell

Usage:
https://github.com/commoditrader/web-app/pull/2048

### Screenshot
Issue SS
image

Fixed
image

### CODE REVIEW

**Review by mtrzesniowski — COMMENTED** [review]
`yarn.lock` not needed

File: export-to-csv.ts [hunk 1]
```diff
@@ -153,12 +153,19 @@ export class ExportToCsv {
...
         for (var i = 0; i < this._data.length; i++) {
+            const keys = Object.keys(this._data[i]);
             let row = "";
+
             for (let keyPos = 0; keyPos < keys.length; keyPos++) {
                 const key = keys[keyPos];
-                row += this._formatData(this._data[i][key]) + this._options.fieldSeparator;
+                const cellData = this._data[i][key];
+
+                if (cellData !== "" && !cellData) {
```
Comment (kubakrupski) [inline export-to-csv.ts]: Prevent to render undefined when cell is empty and allow to render empty string ""

File: export-to-csv.ts [hunk 2]
```diff
@@ -153,12 +153,19 @@ export class ExportToCsv {
      * Create Body
      */
     private _getBody() {
-        const keys = Object.keys(this._data[0]);
```
Comment (kubakrupski) [inline export-to-csv.ts:156]: In that library the first row of passed CSV data says how many columns can have rows, I have changed that behaviour which prevents us to hack the data

File: package.json [hunk 3]
```diff
@@ -4,12 +4,14 @@
...
-        "url": "https://github.com/alexcaza/export-to-csv.git"
+        "url": "git+https://github.com/commoditrader/export-to-csv.git#fd88cd5"
     },
     "scripts": {
         "dev": "tsc --watch",
         "test": "tsc && karma start",
-        "prepublishOnly": "tsc"
+        "prepublishOnly": "tsc",
+        "build": "tsc",
+        "postinstall": "npm run build"
```
Comment (kubakrupski) [inline package.json]: Allow to run that command on post-install phase, npm can run directly  `npm run tsc` but can do that using the other scripts for some reason. There is a topic about that on the WEB

File: export-to-csv.ts (hunk 1 above)
Comment (mtrzesniowski) [reply export-to-csv.ts]: won't we miss the field separator if you do `continue`?

why not go with this approach in `_formatData`?
```ts
if (!data ) {
  return "";
}
```

like in this PR:
https://github.com/alexcaza/export-to-csv/pull/26

File: .gitignore [hunk 4]
```diff
@@ -12,4 +12,5 @@ Thumbs.db
 *.iml
 test/
 coverage/
-build/
\ No newline at end of file
+build/
+yarn.lock
```
Comment (alysakowski) [inline .gitignore:16]: ignore yarn.lock?

**Review by alysakowski — COMMENTED** [review]
one comment and please see matt's comment as well

**Pull request merged**